Flaky test: TestReconnectStateSync intermittently fails

## Description

`TestReconnectStateSync` (`integration_test.go:781`) intermittently fails with:

```
--- FAIL: TestReconnectStateSync (1.60s)
    Error Trace: integration_test.go:781
    Error:       Should be true
```

## Context

Observed during PR #856 CI run ([run 23078768896](https://github.com/NickBorgers/home-automation/actions/runs/23078768896)). Re-running the same job passed, confirming this is a flaky test — likely a timing/race condition in the reconnect state sync logic.

This is particularly relevant because PR #856 switches wait helpers from `assert.Eventually` to `require.Eventually`. The old `assert.Eventually` would silently swallow these failures (the test would appear to pass). With `require.Eventually`, flaky tests like this will surface as real failures — which is the correct behavior, but means we need to fix the underlying flakiness.

## Expected behavior

`TestReconnectStateSync` should pass reliably on every run.

## Investigation hints

- Check if the test has adequate wait timeouts for state sync after reconnect
- Look for race conditions in the reconnect path (missing mutex, state read before write completes)
- Consider whether `TEST_WAIT_TIMEOUT` (10s in CI) is sufficient for reconnect scenarios

## Files to review
- `homeautomation-go/test/integration/integration_test.go` (line 781)
- Related reconnect logic in `internal/ha/`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/homeautomation-go/test/integration/integration_test.go
+++ b/homeautomation-go/test/integration/integration_test.go
@@ -785,9 +785,12 @@ func TestReconnectStateSync(t *testing.T) {
 
 	// CRITICAL: Verify state was synced after reconnect
 	// The state manager should now see Nick as home (changed while disconnected)
-	nickHome, err := manager.GetBool("isNickHome")
-	require.NoError(t, err)
-	assert.True(t, nickHome, "Nick should be home after reconnect sync - state changed while disconnected")
+	// Note: callbackInvoked is set before SyncFromHA() completes, so we must poll
+	// until the state is actually synced rather than asserting immediately.
+	waitForCondition(t, func() bool {
+		nickHome, err := manager.GetBool("isNickHome")
+		return err == nil && nickHome
+	}, "Nick should be home after reconnect sync - state changed while disconnected")
 
 	t.Log("✅ State successfully synchronized after reconnection!")
 

